### PR TITLE
YLP-1740 Make HTTP calls to back instead of computing data in the front

### PR DESCRIPTION
**Pull request by TimCanu**

### CODE REVIEW

File: packages/yourloops/lib/team/hook.tsx [hunk 1]
```diff
@@ -83,22 +78,43 @@ function TeamContextImpl(): TeamContext {
...
   }
 
-  const getMapUsers = (): Map<string, TeamUser> => {
-    const users = new Map<string, TeamUser>()
-    for (const team of teams) {
-      for (const member of team.members) {
-        if (!users.has(member.user.userid)) {
-          users.set(member.user.userid, member.user)
+  const fetchTeams = useCallback(() => {
+    TeamUtils.loadTeams(user)
```
Comment (fatb38) [inline packages/yourloops/lib/team/hook.tsx:82]: Could you rewrite it with async/await ? (for code consistency)

File: packages/yourloops/lib/patient/hook.tsx [hunk 2]
```diff
@@ -33,28 +33,43 @@ import { Team, useTeam } from '../team'
...
+  const { refresh: refreshTeams } = useTeam()
   const { user, getFlagPatients, flagPatient } = useAuth()
 
   const [patients, setPatients] = useState<Patient[]>([])
   const [initialized, setInitialized] = useState<boolean>(false)
+  const [refreshInProgress, setRefreshInProgress] = useState<boolean>(false)
   const [errorMessage, setErrorMessage] = React.useState<string | null>(null)
 
+  const fetchPatients = useCallback(() => {
+    PatientUtils.computePatients().then(computedPatients => {
```
Comment (fatb38) [inline packages/yourloops/lib/patient/hook.tsx:55]: Could you rewrite it with async/await ? (for code consistency)

File: packages/yourloops/lib/team/hook.tsx [hunk 3]
```diff
@@ -41,21 +40,17 @@ import { useNotification } from '../notifications/hook'
 import { Team, TeamContext, TeamMember, TeamUser } from './models'
 import TeamApi from './team-api'
 import TeamUtils from './utils'
-import { notificationConversion } from '../notifications/utils'
 import { CircularProgress } from '@material-ui/core'
 
 const log = bows('TeamHook')
 const ReactTeamContext = React.createContext<TeamContext>({} as TeamContext)
-/** hackish way to prevent 2 or more consecutive loading */
-let lock = false
```
Comment (fatb38) [inline packages/yourloops/lib/team/hook.tsx:50]: 👌🏻

File: packages/yourloops/lib/team/hook.tsx [hunk 4]
```diff
@@ -252,56 +217,21 @@ function TeamContextImpl(): TeamContext {
...
-    refresh(true)
+    refresh()
   }
 
-  const initHook = (): void => {
-    if (initialized || lock || !notificationHook.initialized) {
-      return
+  React.useEffect(() => {
+    if (!initialized && notificationHook.initialized) {
+      log.info('init')
```
Comment (fatb38) [inline packages/yourloops/lib/team/hook.tsx:225]: what do you think about removing theses logs, (useless I think)

File: packages/yourloops/lib/team/hook.tsx (hunk 1 above)
Comment (TimCanu) [reply packages/yourloops/lib/team/hook.tsx:82]: Hummm I thought useEffect did not handle await very well... (not considering IIFE)

File: packages/yourloops/lib/team/hook.tsx (hunk 4 above)
Comment (TimCanu) [reply packages/yourloops/lib/team/hook.tsx:225]: I agree

File: packages/yourloops/lib/team/hook.tsx (hunk 1 above)
Comment (TimCanu) [reply packages/yourloops/lib/team/hook.tsx:82]: I won't do it. The cost is too big and annoying for how little the benefit is

**Review by fatb38 — APPROVED** [review]

File: packages/yourloops/lib/team/hook.tsx (hunk 4 above)
Comment (fatb38) [reply packages/yourloops/lib/team/hook.tsx]: Could you remove all logs in this file (I just discover logs in other methods)

**Review by fatb38 — CHANGES_REQUESTED** [review]
Just discover a useEffect leak when leaving a team. Could be fixed

**Comment by TimCanu** [comment]
@fatb38 Are you talking about this?
```
Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function.
    in LeaveTeamButton (created by TeamInformation)
    in div (created by TeamInformation)
    in div (created by TeamInformation)
    in div (created by TeamInformation)
    in TeamInformation (created by TeamDetailsPage)
    in div (created by TeamDetailsPage)
    in div (created by TeamDetailsPage)
    in div (created by Styled(MuiBox))
    in Styled(MuiBox) (created by TeamDetailsPage)
    in div (created by Styled(MuiBox))
    in Styled(MuiBox) (created by TeamDetailsPage)
    in div (created by Styled(MuiBox))
    in Styled(MuiBox) (created by TeamDetailsPage)
    in TeamDetailsPage (created by Context.Consumer)
    in Route (created by PatientLayout)
```
Because I don't see anything else and this is also on dblp branch (and I don't think that it is related to the team hook)

**Review by fatb38 — APPROVED** [review]

**Pull request merged**